spring-integration-core/src/main/java/org/springframework/integration/transformer/FromProtobufTransformer.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2023-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.transformer;
+
+import org.springframework.beans.factory.BeanClassLoaderAware;
+import org.springframework.expression.EvaluationContext;
+import org.springframework.expression.Expression;
+import org.springframework.integration.context.IntegrationContextUtils;
+import org.springframework.integration.expression.FunctionExpression;
+import org.springframework.integration.transformer.support.ProtoHeaders;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.converter.ProtobufMessageConverter;
+import org.springframework.util.Assert;
+import org.springframework.util.ClassUtils;
+
+/**
+ * A Protocol Buffer transformer to instantiate {@link com.google.protobuf.Message} objects from {@code byte[]}.
```

Comment:
Is that still correct to say "from `byte[]`" since we have just talked about JSON which is accepted only as as String?
Shouldn't we improve this Javadocs respectively with a `ProtobufMessageConverter` logic?
And still mention that we fully rely on that converter internally.